include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h

```diff
@@ -1242,18 +1244,27 @@ namespace internal
       // check if FE is the same
       AssertDimension(fe_coarse.n_dofs_per_cell(), fe_fine.n_dofs_per_cell());
 
+
+      const bool is_feq =
+        fe_fine.n_base_elements() == 1 &&
+        (dynamic_cast<const FE_Q<dim> *>(&fe_fine.base_element(0)) != nullptr);
+
       // number of dofs on coarse and fine cells
       transfer.schemes[0].n_dofs_per_cell_coarse =
         transfer.schemes[0].n_dofs_per_cell_fine =
           transfer.schemes[1].n_dofs_per_cell_coarse =
             fe_coarse.n_dofs_per_cell();
       transfer.schemes[1].n_dofs_per_cell_fine =
-        fe_coarse.n_dofs_per_cell() * GeometryInfo<dim>::max_children_per_cell;
+        is_feq ? (fe_fine.n_components() *
+                  Utilities::pow(2 * fe_fine.degree + 1, dim)) :
+                 (fe_coarse.n_dofs_per_cell() *
+                  GeometryInfo<dim>::max_children_per_cell);
 
       // degree of FE on coarse and fine cell
       transfer.schemes[0].degree_coarse   = transfer.schemes[0].degree_fine =
         transfer.schemes[1].degree_coarse = fe_coarse.degree;
-      transfer.schemes[1].degree_fine     = fe_coarse.degree * 2 + 1;
+      transfer.schemes[1].degree_fine =
+        is_feq ? (fe_coarse.degree * 2) : (fe_coarse.degree * 2 + 1);
```

Comment:
Without further comment, this appears to be inconsistent with line 1261 above, as I would expect
```suggestion
        is_feq ? (fe_coarse.degree * 2 + 1) : (fe_coarse.degree * 2 + 2);
```
But I think I understand things that this is not actually a degree and you add one to the degree in the definition of the number of dofs along each direction. I think it might make sense to rename the variable `degree` to something more expressive, thus avoiding the confusion, or to add a comment how the name `degree` is to be interpreted and used in the functions above.

Reply:
I see your point related to `degree` but I would keep it for now, since it has been like this for more than a year. In a follow-up PR, I can replace degree by n_points_1D.